pkg/kubelet/kubelet.go

```diff
@@ -2076,6 +2077,15 @@ func (kl *Kubelet) SyncPod(ctx context.Context, updateType kubetypes.SyncPodType
 				kl.statusManager.SetPodResizeInProgressCondition(pod.UID, v1.PodReasonError, r.Message, false)
 			}
 		}
+
+		if errors.Is(r.Error, images.ErrImagePull) {
+			hasPullFailures = true
+		}
+	}
+
+	// If we have any image pull errors, report any failed pull secrets
+	if hasPullFailures && len(failedPullSecretNames) > 0 {
+		kl.recorder.Eventf(pod, v1.EventTypeWarning, "FailedToRetrieveImagePullSecret", "Unable to retrieve some image pull secrets: %v", failedPullSecretNames)
```

Comment:
The message here is scarier than it used to be. I would be ok to keep it as it was before. Or change it to indicate that those "may be" a problem for image pulling.